x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/SearchableSnapshotDirectory.java

```diff
@@ -444,7 +446,17 @@ public static Directory create(
         );
 
         final LazyInitializable<BlobContainer, RuntimeException> lazyBlobContainer = new LazyInitializable<>(
-            () -> blobStoreRepository.shardContainer(indexId, shardPath.getShardId().id())
+            () -> new FilterBlobContainer(blobStoreRepository.shardContainer(indexId, shardPath.getShardId().id())) {
```

Comment:
It's a bit strange to require callers to make sure to apply this rate limiting, rather than having the repository create a blob container that already includes this. But I don't see an easy way to push this down to the blob container level either, so I'm not asking for any changes here.

Reply:
I agree. I admit I'm also confused that `max_restore_bytes_per_sec` is in fact a limitation of the reading throughput from the repository rather than a writing throughput to the local disk. I'd expect the restore process to use a `RateLimitedIndexOutput` or something like that.